lib/data_update_scripts/20210316091354_move_authentication_settings.rb

```diff
@@ -0,0 +1,42 @@
+module DataUpdateScripts
+  class MoveAuthenticationSettings
+    AUTHENTICATION_SETTINGS = %w[
+      allow_email_password_login
+      allow_email_password_registration
+      apple_client_id
+      apple_key_id
+      apple_pem
+      apple_team_id
+      display_email_domain_allow_list_publicly
+      facebook_key
+      facebook_secret
+      github_key
+      github_secret
+      invite_only_mode
+      require_captcha_for_email_password_registration
+      twitter_key
+      twitter_secret
+    ].freeze
+
+    ATTRIBUTES = %i[var value created_at updated_at].freeze
+
+    def run
+      return if Settings::Authentication.any?
+      
+      SiteConfig.transaction do
+        config_relation = SiteConfig.where(var: AUTHENTICATION_SETTINGS)
+        config_values = config_relation.pluck(*ATTRRIBUTES).map do |values|
+          ATTRIBUTES.zip(values).to_h
+        end
+        Settings::Authentication.insert_all(config_values)
+
+        # This field has a validation we don't want to skip
+        Settings::Authentication.allowed_registration_email_domains =
+          SiteConfig.allowed_registration_email_domains
```

Comment:
`SiteConfig.allowed_registration_email_domains` doesn't exist anymore, so it's not portable. I wonder if we should keep the old fields in `SiteConfig` in this PR and then remove them in a following PR once the migration has been completed on the entire Forem Fleet

Reply:
I re-added this and `authentication_providers`.